File: 프로그래머스/2/87946. 피로도/피로도.js
function solution(k, dungeons) {
    dungeons.sort((a,b)=>(a[0]-a[1])-(b[0]-b[1]));
    let arr = Array.from({length:dungeons.length+1}, ()=> Array(k+1).fill(0));
    for (let i = 0; i < dungeons.length; i++) {
        for (let j = 1; j < k+1; j++) {
            if (dungeons[i][0] > j) arr[i+1][j] = arr[i][j];
            else arr[i+1][j] = Math.max(arr[i][j], 1 + arr[i][j-dungeons[i][1]])
        }
    }
    return arr[dungeons.length][k]
}

File: 프로그래머스/2/87946. 피로도/피로도.py
def solution(k, dungeons):
    dungeons.sort(key=lambda x: (x[0]-x[1]))
    arr = [0]*(k+1)
    for i in range(len(dungeons)):
        for j in range(k, dungeons[i][0]-1, -1):
            arr[j] = max(arr[j], 1 + arr[j-dungeons[i][1]])
    return arr[k]